src/log_batch.rs

```diff
@@ -765,12 +767,62 @@ impl LogBatch {
     }
 
     #[allow(dead_code, unused_variables)]
-    pub(crate) fn sign_checksum(&mut self, file_version: Version, file_id: FileId) {
-        // @TODO: lucasliang
-        // Supplement the details of signing checksum on each LogBatch with the
-        // specific `Version`.
-        unimplemented!();
-        // debug_assert!(matches!(self.buf_state, BufState::Sealed(_, _)));
+    /// Signs a checksum, so-called `signature`, into the `LogBatch`.
+    ///
+    /// The `signature` is both generated by the given `[Version]` and
+    /// `[FileId]`. That is, the final checksum of each `LogBatch`
+    /// consists of this `signature` and the `checksum` of the contents.
+    pub(crate) fn sign_checksum(&mut self, format_version: Version, file_id: FileId) {
+        debug_assert!(matches!(self.buf_state, BufState::Sealed(_, _)));
+        if !self.is_empty() {
+            match format_version {
+                Version::V1 => {
+                    // default with `Version::V1`, nothing needs to be done.
+                }
+                Version::V2 => {
+                    // Insert the signature into the encoded bytes.
+                    match self.buf_state {
+                        BufState::Sealed(header_offset, len) => {
+                            // Rewrite checksum of compressed / raw content of LogBatch
```

Comment:
Only need to sign item batch. We only retrieve item batch during recovery.

Reply:
Got it.